### PR TITLE
File Tree Diff: initial implementation

### PR DESCRIPTION
**Pull request by humitos**
Added a new addon to show all the filenames and URLs that were added, deleted or modified in the current PR compared to the LATEST version.

Screenshot_2024-11-07_12-44-02

I'm rendering this data in a new notification decoupled from the existing one on purpose. I don't want to entangle the code of these two different addons for now, in particular while we are under development/testing. I also see that in the future they may have different UI elements, or the data will be integrated in a more bigger UI element (eg. toolbar) or similar.

We can discuss more how this addon will grow and where is a good place to add this data and make that work in a following PR. For now, I want something that we can start testing internally in our own projects.

### Examples

Screenshot_2024-11-07_14-15-37

Screenshot_2024-11-07_14-35-41

Screenshot_2024-11-07_14-51-06

Closes https://github.com/readthedocs/addons/issues/409
Requires https://github.com/readthedocs/readthedocs.org/pull/11749

### CODE REVIEW

**Comment by humitos** [comment]
### Usage example (together with docdiff)

Peek 2024-11-07 15-13

**Review by ericholscher — APPROVED** [review]
This looks like a great start. I'm :+1: on shipping it overall. I'm not well versed on the JS style other than my silly comment below :)

File: src/filetreediff.css [hunk 1]
```diff
@@ -0,0 +1,86 @@
+:host > div {
+  margin: 1rem 0rem;
+  padding-top: 1rem;
+  padding-bottom: 1rem;
+  overflow: auto;
+  border-radius: 0.5rem;
+  font-family: "Lato";
```
Comment (agjohnson) [inline src/filetreediff.css:7]: You should use a more permissive font family, as Lato isn't hosted by the Addons and it isn't commonly installed for end users. This is an okay default:

https://github.com/readthedocs/addons/blob/0f025e3bf604f96506aebf2af120b4c20d02fde8/src/notification.css#L14C1-L21C5

**Review by agjohnson — APPROVED** [review]
This all looks great! I think this is good to start testing.

We have had some related conversations recently that would change this work -- using attributes over classes, naming of this feature -- but don't think these need to hold up our internal testing.

**Pull request merged**